modules/openthread/platform/radio_spinel.cpp

```diff
@@ -0,0 +1,586 @@
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->SetCoexEnabled(aEnabled);
+}
+
+bool otPlatRadioIsCoexEnabled(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->IsCoexEnabled();
+}
+
+otError otPlatRadioGetCoexMetrics(otInstance *aInstance, otRadioCoexMetrics *aCoexMetrics)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+
+	otError error = OT_ERROR_NONE;
+
+	VerifyOrExit(aCoexMetrics != NULL, error = OT_ERROR_INVALID_ARGS);
+
+	error = psRadioSpinel->GetCoexMetrics(*aCoexMetrics);
+
+exit:
+	return error;
+}
+#endif
+
+#if OPENTHREAD_CONFIG_DIAG_ENABLE
+otError otPlatDiagProcess(otInstance *aInstance, int argc, char *argv[], char *aOutput, size_t aOutputMaxLen)
+{
+	/* Deliver the platform specific diags commands to radio only ncp */
+	OT_UNUSED_VARIABLE(aInstance);
+	char  cmd[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE] = {'\0'};
+	char *cur											   = cmd;
+	char *end											   = cmd + sizeof(cmd);
+
+	for (int index = 0; index < argc; index++)
+	{
+		cur += snprintf(cur, static_cast<size_t>(end - cur), "%s ", argv[index]);
+	}
+
+	return psRadioSpinel->PlatDiagProcess(cmd, aOutput, aOutputMaxLen);
+}
+
+void otPlatDiagModeSet(bool aMode)
+{
+	SuccessOrExit(psRadioSpinel->PlatDiagProcess(aMode ? "start" : "stop", NULL, 0));
+	psRadioSpinel->SetDiagEnabled(aMode);
+
+exit:
+	return;
+}
+
+bool otPlatDiagModeGet(void)
+{
+	return psRadioSpinel->IsDiagEnabled();
+}
+
+void otPlatDiagTxPowerSet(int8_t aTxPower)
+{
+	char cmd[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
+
+	snprintf(cmd, sizeof(cmd), "power %d", aTxPower);
+	SuccessOrExit(psRadioSpinel->PlatDiagProcess(cmd, NULL, 0));
+
+exit:
+	return;
+}
+
+void otPlatDiagChannelSet(uint8_t aChannel)
+{
+	char cmd[OPENTHREAD_CONFIG_DIAG_CMD_LINE_BUFFER_SIZE];
+
+	snprintf(cmd, sizeof(cmd), "channel %d", aChannel);
+	SuccessOrExit(psRadioSpinel->PlatDiagProcess(cmd, NULL, 0));
+
+exit:
+	return;
+}
+
+void otPlatDiagRadioReceived(otInstance *aInstance, otRadioFrame *aFrame, otError aError)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	OT_UNUSED_VARIABLE(aFrame);
+	OT_UNUSED_VARIABLE(aError);
+}
+
+void otPlatDiagAlarmCallback(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+}
+#endif /* OPENTHREAD_CONFIG_DIAG_ENABLE */
+
+uint32_t otPlatRadioGetSupportedChannelMask(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->GetRadioChannelMask(false);
+}
+
+uint32_t otPlatRadioGetPreferredChannelMask(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->GetRadioChannelMask(true);
+}
+
+otRadioState otPlatRadioGetState(otInstance *aInstance)
+{
+	OT_UNUSED_VARIABLE(aInstance);
+	return psRadioSpinel->GetState();
+}
+
+void otPlatRadioSetMacKey(otInstance			 *aInstance,
+						  uint8_t				 aKeyIdMode,
+						  uint8_t				 aKeyId,
+						  const otMacKeyMaterial *aPrevKey,
+						  const otMacKeyMaterial *aCurrKey,
+						  const otMacKeyMaterial *aNextKey,
+						  otRadioKeyType		  aKeyType)
```

Comment:
This looks copletely misalligned too:
![image](https://github.com/user-attachments/assets/84fdfd0e-b59a-41a8-8c64-fa141a23223e)

Reply:
Resolved